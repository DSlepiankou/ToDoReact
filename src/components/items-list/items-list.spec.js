import  configureStore  from 'redux-mock-store';
import { Provider } from "react-redux";
import { render } from '@testing-library/react';
import ItemsList from './items-list';

const mockStore = configureStore([]);

const testStore = mockStore({
    toDoList:{
        elements:[
            {content:"test item 1", completed: false, id: 1},
            {content:"test item 2", completed: true, id: 2},
            {content:"test item 3", completed: false, id: 3}]
    }
})

describe('Item list component testing', () => {

    const itemsListComponent = render(
        <Provider store={testStore}>
            <ItemsList/>
        </Provider>
    )

    it('renders properly', () => {
        expect(itemsListComponent).toMatchSnapshot()
    });

});