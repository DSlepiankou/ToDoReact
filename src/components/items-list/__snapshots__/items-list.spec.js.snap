// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item list component testing renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list-group todo-list"
      >
        <li
          class="list-group-item"
        >
          <span
            class="todo-list-item"
          >
            <div>
              <input
                class="checkbox-circle"
                type="checkbox"
              />
              test item 1
            </div>
            <button
              class="float-right delete-btn"
              type="button"
            >
              <i
                class="bi bi-trash"
              />
            </button>
          </span>
        </li>
        <li
          class="list-group-item"
        >
          <span
            class="todo-list-item"
          >
            <div>
              <input
                checked=""
                class="checkbox-circle"
                type="checkbox"
              />
              test item 2
            </div>
            <button
              class="float-right delete-btn"
              type="button"
            >
              <i
                class="bi bi-trash"
              />
            </button>
          </span>
        </li>
        <li
          class="list-group-item"
        >
          <span
            class="todo-list-item"
          >
            <div>
              <input
                class="checkbox-circle"
                type="checkbox"
              />
              test item 3
            </div>
            <button
              class="float-right delete-btn"
              type="button"
            >
              <i
                class="bi bi-trash"
              />
            </button>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list-group todo-list"
    >
      <li
        class="list-group-item"
      >
        <span
          class="todo-list-item"
        >
          <div>
            <input
              class="checkbox-circle"
              type="checkbox"
            />
            test item 1
          </div>
          <button
            class="float-right delete-btn"
            type="button"
          >
            <i
              class="bi bi-trash"
            />
          </button>
        </span>
      </li>
      <li
        class="list-group-item"
      >
        <span
          class="todo-list-item"
        >
          <div>
            <input
              checked=""
              class="checkbox-circle"
              type="checkbox"
            />
            test item 2
          </div>
          <button
            class="float-right delete-btn"
            type="button"
          >
            <i
              class="bi bi-trash"
            />
          </button>
        </span>
      </li>
      <li
        class="list-group-item"
      >
        <span
          class="todo-list-item"
        >
          <div>
            <input
              class="checkbox-circle"
              type="checkbox"
            />
            test item 3
          </div>
          <button
            class="float-right delete-btn"
            type="button"
          >
            <i
              class="bi bi-trash"
            />
          </button>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
